projects/rocprofiler-sdk/source/lib/rocprofiler-sdk/agent.hpp

```diff
@@ -123,5 +123,9 @@ construct_agent_cache(::HsaApiTable* table);
 
 void
 internal_refresh_topology();  // only for internal testing
+
+// Null/invalid agent ID constant
+constexpr rocprofiler_agent_id_t null_agent_id = rocprofiler_agent_id_t{.handle = 0};
```

Comment:
```suggestion
constexpr auto null_agent_id = rocprofiler_agent_id_t{.handle = 0};
```
Avoids redundant use of `rocprofiler_agent_id_t`. 

In general, any internal static library without rocprofiler-sdk in its name should not be including internal headers from libraries with rocprofiler-sdk in its name.